client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/impl/iteration/RemoteInnerPublisherHandler.java

```diff
@@ -89,7 +90,13 @@ protected long handleNextResponse(IterationNextResponse<K, E> nextResponse, Map.
       if (!nextResponse.hasMore()) {
          // server doesn't clean up when complete
          sendCancel(target);
-         publisher.completeSegments(target.getValue());
+         // Don't complete the segments if it was an invalid iteration
+         if (nextResponse.getStatus() != HotRodConstants.INVALID_ITERATION) {
```

Comment:
This is the fix here. While shutting down a server can send back invalid iteration responses with no values.

Now when running the test I can see trace messages like 

08:16:07,921 TRACE (DistFailOverRemoteIteratorTest-Client-Async-1-2:[]) [o.i.c.h.i.i.RemoteInnerPublisherHandler] Invalid iteration response, so must retry segments {91 201-210 224-230}

that would have caused this to fail as per the flaky test.